### PR TITLE
Reorder habitat build resource to cleanup the studio before we try running the build

### PR DESCRIPTION
**Pull request by chef-delivery**
This Pull Request was opened by Chef Automate user Elliott Davis

* removes cleanup property

Ready to merge? View this change in Chef Automate and click the Approve button: https://delivery.chef.co/e/chef/#/organizations/Delivery-Build-Cookbooks/projects/habitat-build/changes/9b04853d-af17-4acc-8327-0aacb7086912

### CODE REVIEW

File: spec/unit/resources/hab_build_spec.rb [hunk 1]
```diff
@@ -3,42 +3,24 @@
 
 describe 'test::build' do
   describe ':build action' do
-    let(:cleanup) { false }
-
     let(:chef_run) do
```
Comment (phiggins) [inline spec/unit/resources/hab_build_spec.rb]: FWIW, this `let` used to be `cached` before it needed to take the `cleanup` parameter.

**Comment by elliott-davis** [comment]
@phiggins got that fixed up

File: resources/hab_build.rb [hunk 2]
```diff
@@ -63,6 +62,13 @@ def last_build_env
 end
 
 action :build do
+  # Delete studio before we before we run. Run in Hab doesn't do this for us
+  # but build does.
+  execute 'remove-studio' do
+    command "sudo -E #{hab_binary} studio -r #{hab_studio_path} rm"
+    live_stream live_stream
```
Comment (rhass) [inline resources/hab_build.rb]: Should this be `live_stream new_resource.live_stream`?

Comment (elliott-davis) [inline resources/hab_build.rb]: Totally right! Good catch!

**Pull request merged**